llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -3304,6 +3304,13 @@ X86TargetLowering::preferredShiftLegalizationStrategy(
                                                             ExpansionFactor);
 }
 
+bool X86TargetLowering::isDesirableToCommuteWithShift(
+    const SDNode *N, CombineLevel Level) const {
+  if (N->getOperand(0)->hasOneUse())
+    return true;
```

Comment:
This doesn't look like the hasOneUse is being applied to the same operand as the original?

Reply:
+1